gitbot.js

```diff
@@ -3,18 +3,28 @@ const subdirs = require('subdirs');
 const isGit = require('is-git');
 const gitlog = require('gitlog');
 const async = require("async");
-const getGitUsername = require('git-user-name');
+const { exec } = require("child_process");
 
-function getGitUser() {
-  try {
-    return getGitUsername();
-  } catch(err) {
-    console.error(`ERROR reading git-config.
+function craftGetGitUserErrorMessage(error, stdout) {
+  return `ERROR reading git-config.
       Use e.g. 'git config --global user.name "Mona Lisa"'.
       See 'man git config' for further information.
-    `);
-    return process.exit(1);
-  }
+      \n${error.message}\n${stdout}\n`;
+}
+
+async function getGitUser() {
+  return new Promise((resolvePromise, rejectPromise) => {
+    exec('git config --get user.name', (error, stdout) => {
+
+      if (error || stdout.trim() === '') {
+        rejectPromise(craftGetGitUserErrorMessage(error, stdout));
```

Comment:
Because the function `craftGetGitUserErrorMessage` is used no where else I would recommend embedding its code here directly.

Reply:
My motivation for splitting out that function was indeed not reusability, but readability. That way, both `getGitUser` and `craftGetGitUserErrorMessage` only have a single responsibility.